### PR TITLE
[ST] improve/fix fails in KafkaRollerST by adding clients 

### PR DESCRIPTION
**Pull request by henryZrncik**
### Type of change

- Enhancement 
- Refactoring

### Description

addition of extra topic creation and data production consumption before scaling up (why: make sure consumer offset topic is created before scaling up so its distribution across brokers won't prevent scaling down later)

addition of extra production and consumption into next topic with more replicas simulating real work with topic which would later have partition removed. This time also actually provides more time to have replicas distributed across brokers since scaling up to start scaling down. 

### Checklist

- [x] Write tests
- [x] Make sure all tests pass

### CODE REVIEW

**Review by jankalinic — APPROVED** [review]
LGTM, lets try it on azp.

**Comment by henryZrncik** [comment]
@strimzi-ci run tests --testcase=KafkaRollerST

**Comment by jankalinic** [comment]
@strimzi-ci run tests --cluster-type=minikube --cluster-version=1.27 --testcase=KafkaRollerST#testKafkaDoesNotRollsWhenTopicIsUnderReplicated

File: systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/KafkaRollerST.java [hunk 1]
```diff
@@ -94,29 +122,44 @@ void testKafkaDoesNotRollsWhenTopicIsUnderReplicated(ExtensionContext extensionC
...
         assertThat(events, hasAllOfReasons(Scheduled, Pulled, Created, Started));
 
+        clients = new KafkaClientsBuilder()
+            .withProducerName(testStorage.getProducerName())
+            .withConsumerName(testStorage.getConsumerName())
+            .withBootstrapAddress(KafkaResources.plainBootstrapAddress(testStorage.getClusterName()))
+            .withTopicName(topicNameWith4Replicas)
+            .withMessageCount(testStorage.getMessageCount())
+            .withNamespaceName(testStorage.getNamespaceName())
+            .build();
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/KafkaRollerST.java:141]:
```suggestion
        clients = new KafkaClientsBuilder(clients)
            .withTopicName(topicNameWith4Replicas)
            .build();
```

**Review by im-konge — APPROVED** [review]
LGTM, just one nit. Thanks

File: systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/KafkaRollerST.java (hunk 1 above)
Comment (henryZrncik) [reply systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/KafkaRollerST.java]: good catch, thanks

**Comment by henryZrncik** [comment]
azure regression pipeline (triggered manually) is all green.

**Review by see-quick — APPROVED** [review]

**Comment by scholzj** [comment]
* There is no record of regression tests being run - especially for KRaft. Running them manually like you do it makes it unclear at what point with what code was it run. So please do not do it!
* KRaft does not support scaling of controller (or mixed) nodes, so likely this test should be disabled there as well.

**Comment by henryZrncik** [comment]
@strimzi-ci run tests --testcase=KafkaRollerST

**Pull request merged**